content/blog/2023/07/31/2023-07-31-documentation-transition-to-java-17.adoc

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title: Java 17 usage in Jenkins.io Documentation
+tags:
+- java
+- linux
+- jenkins
+- documentation
+authors:
+- kmartens27
+opengraph:
+  image: /images/post-images/2023/07/28/jenkins-supports-java-17.png
+links:
+discourse: true
+---
+
+image:/images/post-images/2023/07/28/jenkins-supports-java-17.png[Jenkins supports Java 17., 600]
+
+Jenkins has fully supported Java 17 since the 2.357 weekly release and 2.361.1 Long Term Support release.
+For this reason, we have transitioned to using Java 17 in the Jenkins user handbook.
+This includes the link:/doc/book/installing/[installation guides], link:/doc/book/pipeline/[Pipeline documentation], and link:/doc/book/pipeline/syntax/[Pipeline syntax].
+Refer to the link:https://github.com/jenkins-infra/jenkins.io/issues/6310[GitHub issue] for additional information and insight.
+
+== Background
+
+Debian 12 was released on June 10, 2023, and it no longer delivers OpenJDK 11.
+Due to this change in Debian, we have transitioned to using Java 17 in the documentation.
+The idea is this will encourage users to upgrade to Java 17, which has been fully supported since September 2022.
+Aligning with the actively supported Java version will also help prevent user confusion or frustration when trying to follow installation and usage guides.
+
+== Advantages of Java 17
+
+Java 17 improves upon Java 11 by including several new features that were not previously available.
+Java 17 provides more functional testing and tools to help users improve their workflow and capabilities.
+Including features that provide more extensive development options, such as:
+
+* link:https://docs.oracle.com/en/java/javase/17/text-blocks/index.html[Text blocks]
+* link:https://openjdk.org/jeps/406[Pattern matching for switch expressions]
+* link:https://openjdk.org/jeps/409[Sealed classes]
+* link:https://docs.oracle.com/en/java/javase/17/docs/api/jdk.incubator.foreign/jdk/incubator/foreign/package-summary.html[Foreign memory access]
+* link:https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/Period.html[Day period support]
+* link:https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/CompactNumberFormat.html[Compact number formatting support]
+Read more about these features and how they can be used in the link:https://www.oracle.com/java/technologies/javase/17-relnote-issues.html[Java 17 release notes].
+
```

Comment:
I'd remove the list of Java 17 advantages because they are unlikely to benefit Jenkins users.  They may benefit Jenkins maintainers, but they are not the primary audience for the blog post.

The list is impressive (more than I realized), but it is a developer centered list rather than a user centered list.
```suggestion
```

Reply:
that makes total sense @MarkEWaite, I'll remove the list in that case. Thanks for the insight!